Missing dependencies for plugins

Currently, the `plugins` tag fails to run volatility:
```
$ docker run --rm -v $(pwd):/data:ro blacktop/volatility:plugins --plugins=/plugins --info                                                   

Volatility Foundation Volatility Framework 2.6
*** Failed to import volatility.plugins.community.YingLi.python_strings (ImportError: No module named YingLi.python_strings)
ERROR   : volatility.debug    : Please install DPAPIck library: https://bitbucket.org/jmichel/dpapick
*** Failed to import volatility.plugins.community.StanislasLejay.linux.get_profile (ImportError: No module named linux.get_profile)
*** Failed to import volatility.plugins.community.FrancescoPicasso.mimikatz (AttributeError: 'module' object has no attribute 'ULInt32')
*** Failed to import volatility.plugins.community.AlexanderTarasenko.windbg (ImportError: No module named pykd)
*** Failed to import volatility.plugins.community.TranVienHa.osint (ImportError: No module named socks)
```

I guess as the Community repository evolves, new dependencies are added.

You need to hardcode a tag/commit in the Dockerfile to make sure you install all the dependencies in the build. I will try to create a PR.

Thx for the work maintaining this :-)

If you do get some extra time it might also be cool to decide which plugins make sense to install and which ones do not. for example. `windbg` might only make sense on Windows. 

I am also confused by the error output:

```
Volatility Foundation Volatility Framework 2.6
*** Failed to import volatility.plugins.community.YingLi.python_strings (ImportError: No module named YingLi.python_strings)
ERROR   : volatility.debug    : Please install DPAPIck library: https://bitbucket.org/jmichel/dpapick
*** Failed to import volatility.plugins.community.StanislasLejay.linux.get_profile (ImportError: No module named linux.get_profile)
*** Failed to import volatility.plugins.community.FrancescoPicasso.mimikatz (AttributeError: 'module' object has no attribute 'ULInt32')
*** Failed to import volatility.plugins.community.AlexanderTarasenko.windbg (ImportError: No module named pykd)
*** Failed to import volatility.plugins.community.TranVienHa.osint (ImportError: No module named socks)
make: *** [test] Error 1
```
 
Isn't `linux.get_profile` a part of volatility proper?

--- a/README.md
+++ b/README.md
@@ -66,19 +66,27 @@ Offset(V)  Name                    PID   PPID   Thds     Hnds   Sess  Wow64 Star
 
 ### Documentation
 
-##### To use additional plugins
+##### Plugins tag
+
+This image includes and enables by default plugins from:
+
+- [contrib/plugins](https://github.com/volatilityfoundation/volatility/tree/master/contrib/plugins) from the main volatility package.
+- [Volatility community repo](https://github.com/volatilityfoundation/community)
+
+###### To use additional plugins
 
 ```bash
 $ docker run --rm -v /path/to/mem:/data:ro \
                   -v /path/to/plugins:/plugins \
-                  blacktop/volatility --plugins=/plugins -f silentbanker.vmem custom_plugin
+                  blacktop/volatility -f silentbanker.vmem custom_plugin
 ```
-
+> **NOTE:** This will disable all other non-core plugins (contrib and community)
 > **NOTE:** See [Specifying Additional Plugin Directories](https://github.com/volatilityfoundation/volatility/wiki/Volatility%20Usage#specifying-additional-plugin-directories)
 
-Add the following to your bash or zsh profile
 
 ##### Use **blacktop/volatility** like a host binary
+
+Add the following to your bash or zsh profile
 
 ```bash
 alias vol='docker run -it --rm -v $(pwd):/data:ro blacktop/volatility $@'

--- a/w-plugins/Dockerfile
+++ b/w-plugins/Dockerfile
@@ -21,7 +21,7 @@ RUN apk-install -t .build-deps \
   && export PIP_DISABLE_PIP_VERSION_CHECK=on \
   && pip install --upgrade pip wheel \
   && pip install simplejson \
-                 construct \
+                 construct==2.5.5-reupload \
                  openpyxl \
                  haystack \
                  distorm3 \
@@ -29,6 +29,10 @@ RUN apk-install -t .build-deps \
                  ipython \
                  pycoin \
                  pytz \
+                 pysocks \
+                 requests \
+                 dpapick \
+
   && echo "===> Installing Volatility from source..." \
   && cd /tmp \
   && git clone --recursive --branch $VOL_VERSION https://github.com/volatilityfoundation/volatility.git \
@@ -38,6 +42,8 @@ RUN apk-install -t .build-deps \
   && mkdir /plugins \
   && cd /plugins \
   && git clone https://github.com/volatilityfoundation/community.git \
+  && rm -rf community/AlexanderTarasenko/windbg.py \
+  && rm -rf community/MarcinUlikowski/bitlocker.py \
   && mv /tmp/volatility/contrib/plugins contrib \
   && echo " * Installing ioc_writer..." \
   && cd /tmp \
@@ -46,6 +52,17 @@ RUN apk-install -t .build-deps \
   && python setup.py install \
   && rm -rf /tmp/* \
   && apk del --purge .build-deps
+
+# Temporary hotfixes for upstream volatility/community import errors
+# https://github.com/volatilityfoundation/community/pull/19
+RUN cd /plugins/community/YingLi \
+  && touch __init__.py \
+  && cd /plugins/community/StanislasLejay/linux \
+  && touch __init__.py \
+  && cd /plugins/community/DatQuoc \
+  && touch __init__.py \
+  && cd /plugins/community/DimaPshoul \
+  && sed -i 's/import volatility.plugins.malware.callstacks as/import/' malthfind.py
 
 VOLUME ["/data"]
 VOLUME ["/plugins"]

